tests/rptest/tests/cloud_storage_chunk_read_path_test.py

```diff
@@ -0,0 +1,252 @@
+        )
+        self.si_settings.load_context(self.logger, test_context=test_context)
+
+        super().__init__(
+            test_context=test_context,
+            node_prealloc_count=1,
+            num_brokers=3,
+            si_settings=self.si_settings,
+            extra_rp_conf={'cloud_storage_cache_chunk_size': 1024 * 256})
+
+        self.scale = Scale(test_context)
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.topics = (TopicSpec(name='panda-topic',
+                                 partition_count=1,
+                                 replication_factor=3,
+                                 segment_bytes=self.log_segment_size), )
+
+        self.extra_rp_conf = {'cloud_storage_cache_chunk_size': 1024 * 256}
+        if not self.redpanda.dedicated_nodes:
+            self.extra_rp_conf.update(
+                {'cloud_storage_max_readers_per_shard': 10})
+        self.redpanda.set_extra_rp_conf(self.extra_rp_conf)
+
+    def setup(self):
+        # Do not start redpanda here, let the tests start with custom config options
+        pass
+
+    def teardown(self):
+        self.redpanda.cloud_storage_client.empty_bucket(
+            self.si_settings.cloud_storage_bucket)
+
+    def _set_params_and_start_redpanda(self, **kwargs):
+        if kwargs:
+            self.extra_rp_conf.update(kwargs)
+            self.redpanda.set_extra_rp_conf(self.extra_rp_conf)
+        self.redpanda.start()
+        self._create_initial_topics()
+
+    def _produce_baseline(self, n_segments=20):
+        producer = KgoVerifierProducer(self.test_context,
+                                       self.redpanda,
+                                       self.topic,
+                                       msg_size=1024,
+                                       msg_count=1000 * 200,
+                                       custom_node=self.preallocated_nodes)
+        producer.start()
+        wait_until(
+            lambda: nodes_report_cloud_segments(self.redpanda, n_segments),
+            timeout_sec=180,
+            backoff_sec=3)
+        producer.stop()
+        producer.wait()
+
+        original_snapshot = self.redpanda.storage(
+            all_nodes=True).segments_by_node('kafka',
+                                             self.topic,
+                                             partition_idx=0)
+        for t in self.topics:
+            self.client().alter_topic_config(
+                t.name, TopicSpec.PROPERTY_RETENTION_LOCAL_TARGET_BYTES,
+                self.log_segment_size)
+
+        # Wait for half of the segments to be removed, to exercise the read path
+        # when using the sequential consumer later on.
+        wait_for_removal_of_n_segments(self.redpanda,
+                                       self.topic,
+                                       partition_idx=0,
+                                       n=n_segments // 2,
+                                       original_snapshot=original_snapshot)
```

Comment:
nit: perhaps you could use `wait_for_loca_storage_truncate`. It's slightly more reliable as it does not rely on exact segment counts.

Reply:
changed